File: labs/lab2/lab_python_oop/Circle.py
import math
from lab_python_oop.GeometricFigure import GeometricFigure
from lab_python_oop.FigureColor import Color

class Circle(GeometricFigure):
    name = "Круг"

    def __init__(self, radius, color):
        self.radius = radius
        self.colorObj = Color(color)

    def area(self):
        return math.pi * self.radius ** 2

    def __repr__(self):
        return "Фигура: {0}, радиус: {1}, цвет: {2}, площадь: {3:.2f}".format(
            self.getName(), self.radius, self.colorObj, self.area())

File: labs/lab5_6/Lab2_GeometricFigures/Circle.cs
using System;

namespace Lab2_GeometricFigures
{
    public class Circle : GeometricFigure, IPrint
    {
        public double Radius { get; set; }

        public Circle(double radius)
        {
            if (radius <= 0)
                throw new ArgumentException("Радиус должен быть положительным числом");

            Radius = radius;
        }

        public override double Area()
        {
            return Math.PI * Radius * Radius;
        }

        public override string ToString()
        {
            return $"Круг: Радиус = {Radius:F2}, Площадь = {Area():F2}";
        }

        public void Print()
        {
            Console.WriteLine(ToString());
        }
    }
}
